Missing attribute `allowed_users`in {BitBucket,GitHub,Gitlab,Google}OAuthenticator in 0.13.0 tests

<!-- Thank you for contributing. These HTML comments will not render in the issue, but you can delete them once you've read them if you prefer! -->

### Bug description
Running the tests on 0.13.0 I get  four similar `AttributeError`s.

```
>       assert authenticator.allowed_users == {"blue"}
E       AttributeError: 'BitbucketOAuthenticator' object has no attribute 'allowed_users'
```

<!-- Use this section to clearly and concisely describe the bug. -->

#### Expected behaviour
<!-- Tell us what you thought would happen. -->
Either the attribute should exist on the object, or the assertion should be removed.

#### Actual behaviour
<!-- Tell us what actually happens. -->
The attribute is missing and I cannot find where it would be assigned. It is only asserted for in the tests.

https://github.com/jupyterhub/oauthenticator/search?q=allowed_users

### How to reproduce
<!-- Use this section to describe the steps that a user would take to experience this bug. -->

1. Run the testsuite on the 0.13.0 release from the PyPi tarball

### Your personal set up
<!--
Tell us a little about the system you're using.
Please include information about how you installed,
e.g. are you using a distribution such as zero-to-jupyterhub or the-littlest-jupyterhub.
 -->

 - OS: NixOS Unstable
 <!-- [e.g. ubuntu 20.04, macOS 11.0] -->
 - Version(s):
   - jupyterhub 1.1.0
   - python 3.8.7
 <!-- e.g. jupyterhub --version, python --version --->

- <details><summary><s>Full environment</s></summary>
<!-- For reproduction, it's useful to have the full environment. For example, the output of `pip freeze` or `conda list` --->

```
# paste output of `pip freeze` or `conda list` here
```
</details>

- <details><summary><s>Configuration</s></summary>
<!--
For JupyterHub, especially include information such as what Spawner and Authenticator are being used.
Be careful not to share any sensitive information.
You can paste jupyterhub_config.py below.
To exclude lots of comments and empty lines from auto-generated jupyterhub_config.py, you can do:
    grep -v '\(^#\|^[[:space:]]*$\)' jupyterhub_config.py
-->

```python
# jupyterhub_config.py
```
</details>

- <details><summary>Logs</summary>
<!--
Errors are often logged by jupytehub. How you get logs depends on your deployment.
With kubernetes it might be:

    kubectl get pod # hub pod name starts with hub...
    kubectl logs hub-...
    # or for a single-user server
    kubectl logs jupyter-username

Or the-littlest-jupyterhub:

    journalctl -u jupyterhub
    # or for a single-user server
    journalctl -u jupyter-username
-->

```
=================================== FAILURES ===================================
____________________________ test_deprecated_config ____________________________

caplog = <_pytest.logging.LogCaptureFixture object at 0x7ffff4c4a4f0>

    def test_deprecated_config(caplog):
        cfg = Config()
        cfg.BitbucketOAuthenticator.team_whitelist = ['red']
        cfg.BitbucketOAuthenticator.whitelist = {"blue"}

        log = logging.getLogger("testlog")
        authenticator = BitbucketOAuthenticator(config=cfg, log=log)
        assert (
            log.name,
            logging.WARNING,
            'BitbucketOAuthenticator.team_whitelist is deprecated in BitbucketOAuthenticator 0.12.0, use '
            'BitbucketOAuthenticator.allowed_teams instead',
        ) in caplog.record_tuples

        assert authenticator.allowed_teams == {"red"}
>       assert authenticator.allowed_users == {"blue"}
E       AttributeError: 'BitbucketOAuthenticator' object has no attribute 'allowed_users'

oauthenticator/tests/test_bitbucket.py:101: AttributeError
------------------------------ Captured log call -------------------------------
WARNING  testlog:oauth2.py:361 BitbucketOAuthenticator.team_whitelist is deprecated in BitbucketOAuthenticator 0.12.0, use BitbucketOAuthenticator.allowed_teams instead
____________________________ test_deprecated_config ____________________________

caplog = <_pytest.logging.LogCaptureFixture object at 0x7ffff4ab27f0>

    def test_deprecated_config(caplog):
        cfg = Config()
        cfg.GitHubOAuthenticator.github_organization_whitelist = ["jupy"]
        cfg.Authenticator.whitelist = {"user1"}

        log = logging.getLogger("testlog")
        authenticator = GitHubOAuthenticator(config=cfg, log=log)
        assert (
            log.name,
            logging.WARNING,
            'GitHubOAuthenticator.github_organization_whitelist is deprecated in GitHubOAuthenticator 0.12.0, use '
            'GitHubOAuthenticator.allowed_organizations instead',
        ) in caplog.record_tuples

        assert authenticator.allowed_organizations == {"jupy"}
>       assert authenticator.allowed_users == {"user1"}
E       AttributeError: 'GitHubOAuthenticator' object has no attribute 'allowed_users'

oauthenticator/tests/test_github.py:171: AttributeError
------------------------------ Captured log call -------------------------------
WARNING  testlog:oauth2.py:361 GitHubOAuthenticator.github_organization_whitelist is deprecated in GitHubOAuthenticator 0.12.0, use GitHubOAuthenticator.allowed_organizations instead
____________________________ test_deprecated_config ____________________________

caplog = <_pytest.logging.LogCaptureFixture object at 0x7ffff4aa4760>

    def test_deprecated_config(caplog):
        cfg = Config()
        cfg.GitLabOAuthenticator.gitlab_group_whitelist = {'red'}
        cfg.GitLabOAuthenticator.whitelist = {"blue"}

        log = logging.getLogger("testlog")
        authenticator = GitLabOAuthenticator(config=cfg, log=log)
        assert (
            log.name,
            logging.WARNING,
            'GitLabOAuthenticator.gitlab_group_whitelist is deprecated in GitLabOAuthenticator 0.12.0, use '
            'GitLabOAuthenticator.allowed_gitlab_groups instead',
        ) in caplog.record_tuples

        assert authenticator.allowed_gitlab_groups == {'red'}
>       assert authenticator.allowed_users == {"blue"}
E       AttributeError: 'GitLabOAuthenticator' object has no attribute 'allowed_users'

oauthenticator/tests/test_gitlab.py:275: AttributeError
------------------------------ Captured log call -------------------------------
WARNING  testlog:oauth2.py:361 GitLabOAuthenticator.gitlab_group_whitelist is deprecated in GitLabOAuthenticator 0.12.0, use GitLabOAuthenticator.allowed_gitlab_groups instead
____________________________ test_deprecated_config ____________________________

caplog = <_pytest.logging.LogCaptureFixture object at 0x7ffff4a89190>

    def test_deprecated_config(caplog):
        cfg = Config()
        cfg.GoogleOAuthenticator.google_group_whitelist = {'email.com': ['group']}
        cfg.Authenticator.whitelist = {"user1"}

        log = logging.getLogger("testlog")
        authenticator = GoogleOAuthenticator(config=cfg, log=log)
        assert (
            log.name,
            logging.WARNING,
            'GoogleOAuthenticator.google_group_whitelist is deprecated in GoogleOAuthenticator 0.12.0, use '
            'GoogleOAuthenticator.allowed_google_groups instead',
        ) in caplog.record_tuples

        assert authenticator.allowed_google_groups == {'email.com': ['group']}
>       assert authenticator.allowed_users == {"user1"}
E       AttributeError: 'GoogleOAuthenticator' object has no attribute 'allowed_users'

oauthenticator/tests/test_google.py:129: AttributeError
------------------------------ Captured log call -------------------------------
WARNING  testlog:oauth2.py:361 GoogleOAuthenticator.google_group_whitelist is deprecated in GoogleOAuthenticator 0.12.0, use GoogleOAuthenticator.allowed_google_groups instead
```
</details>

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -60,6 +60,9 @@ jobs:
           - "3.7"
           - "3.8"
           - "3.9"
+        include:
+          - python: "3.9"
+            oldest_dependencies: oldest_dependencies
 
     steps:
       - uses: actions/checkout@v2
@@ -82,7 +85,14 @@ jobs:
       - name: Install Python dependencies
         run: |
           pip install --upgrade pip
-          pip install --upgrade --pre -r test-requirements.txt .
+          pip install --upgrade . -r test-requirements.txt
+          if [ "${{ matrix.oldest_dependencies }}" != "" ]; then
+            # take any dependencies in requirements.txt such as jupyterhub>=1.2
+            # and transform them to jupyterhub==1.2 so we can run tests with the
+            # earliest-supported versions
+            cat requirements.txt | grep '>=' | sed -e 's@>=@==@g' > oldest-requirements.txt
+            pip install -r oldest-requirements.txt
+          fi
           pip freeze
 
       - name: Run tests

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,1 +1,1 @@
-jupyterhub>=0.5
+jupyterhub>=1.2

